src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -186,9 +192,43 @@ protected override async Task OnClientMessageAsync(ConnectionDataMessage connect
             {
                 try
                 {
-                    var payload = connectionDataMessage.Payload;
-                    Log.WriteMessageToApplication(Logger, payload.Length, connectionDataMessage.ConnectionId);
-                    await connection.WriteMessageAsync(payload);
+                    if (connectionDataMessage.IsPartial)
+                    {
+                        var owner = ExactSizeMemoryPool.Shared.Rent((int)connectionDataMessage.Payload.Length);
+                        connectionDataMessage.Payload.CopyTo(owner.Memory.Span);
+                        if (!_bufferingMessages.TryGetValue(connectionDataMessage.ConnectionId, out var list))
```

Comment:
`_ = DispatchMessageAsync(message);`
So there might be mutli-thread issue here?

Reply:
all of the operations are before the first `await`, so there is no multithread issue.